### PR TITLE
Add flow definitions

### PR DESCRIPTION
**Pull request by TrySound**

### CODE REVIEW

File: type-definitions/dom-event-test-flow.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @flow
```
Comment (briancavalier) [inline type-definitions/dom-event-test-flow.js]: We've been using the `type-definitions` folder as the canonical spot for TypeScript files.  We've been keeping flow tests in `test/flow` ([for example](https://github.com/mostjs/core/tree/master/packages/core/test/flow)).

What do you think about moving this file to `test/flow/dom-event-test.js`?

**Review by briancavalier — CHANGES_REQUESTED** [review]
Awesome, @TrySound, thanks.  See question about the location of flow type tests.

File: type-definitions/dom-event-test-flow.js (hunk 1 above)
Comment (TrySound) [reply type-definitions/dom-event-test-flow.js]: Okay. But you already accepted this in most itself
https://github.com/cujojs/most/tree/master/type-definitions

Comment (briancavalier) [reply type-definitions/dom-event-test-flow.js]: Ooops, yes, I must not have been paying attention in that particular case.  Sorry about that.  I'll move them in cujojs/most as well.

File: package.json [hunk 2]
```diff
@@ -12,13 +12,16 @@
...
+    "build:min": "uglifyjs dist/mostDomEvent.js -o dist/mostDomEvent.min.js",
+    "build:flow": "cpy src/index.js.flow dist --rename mostDomEvent.js.flow",
     "prepublish": "npm run build",
     "preversion": "npm run build",
-    "test": "npm run lint && npm run unit-test",
-    "unit-test": "istanbul cover _mocha",
+    "test": "npm run test:lint && npm run test:unit && npm run test:flow",
+    "test:unit": "istanbul cover _mocha",
+    "test:lint": "jsinspect src && jsinspect test && eslint src test --ignore-pattern flow",
+    "test:flow": "flow check",
```
Comment (briancavalier) [inline package.json:24]: Nice, thanks for updating these scripts.

File: package.json [hunk 3]
```diff
@@ -12,13 +12,16 @@
     "type-definitions/dom-event.d.ts"
   ],
   "scripts": {
-    "build": "npm run build-dist && uglifyjs dist/mostDomEvent.js -o dist/mostDomEvent.min.js",
-    "build-dist": "mkdir -p dist && rollup -c && rollup -c rollup.es.js",
-    "lint": "jsinspect src && jsinspect test && eslint src test",
+    "build": "npm run build:dist && npm run build:min && npm run build:flow",
+    "build:dist": "mkdir -p dist && rollup -c && rollup -c rollup.es.js",
+    "build:min": "uglifyjs dist/mostDomEvent.js -o dist/mostDomEvent.min.js",
+    "build:flow": "cpy src/index.js.flow dist --rename mostDomEvent.js.flow",
```
Comment (briancavalier) [inline package.json:18]: I think we also have to copy to `mostDomEvent.es.js.flow`, similar to [most/core](https://github.com/mostjs/core/blob/master/packages/core/package.json#L21).

**Review by briancavalier — CHANGES_REQUESTED** [review]
One minor thing.  Otherwise, looks good.

File: package.json (hunk 3 above)
Comment (TrySound) [reply package.json:18]: module field is supported by both rollup and webpack. There's no reasons to duplicate stuff. Flow will look at main field only.

Comment (briancavalier) [reply package.json:18]: Oh, hmmm, I thought flow fully resolved the js filename first, and then looked for a file of the same name with `.flow` appended.  I'll do a quick test just to convince myself we don't need the "es.js.flow" copy ...

Comment (briancavalier) [reply package.json:18]: I just found [this issue](https://github.com/facebook/flow/issues/3980), which seems to confirm.

**Review by briancavalier — APPROVED** [review]

**Pull request merged**